apps/desktop-wallet/src/features/historicChart/useHistoricData.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AddressHash, CHART_DATE_FORMAT, client, ONE_DAY_MS, TokenHistoricalPrice } from '@alephium/shared'
+import { ALPH } from '@alephium/token-list'
+import { explorer } from '@alephium/web3'
+import { useQueries, useQuery } from '@tanstack/react-query'
+import dayjs from 'dayjs'
+
+import { useAppSelector } from '@/hooks/redux'
+
+const HISTORY_QUERY_KEY = 'history'
+const DAILY = explorer.IntervalType.Daily
+
+type Timestamp = string
+type Amount = string
+
+const useHistoricData = () => {
+  const allAddressHashes = useAppSelector((s) => s.addresses.ids as AddressHash[])
+  const currency = useAppSelector((s) => s.settings.fiatCurrency).toLowerCase()
+
+  const { data: alphPriceHistory, isPending: isPendingAlphPriceHistory } = useQuery({
+    queryKey: [HISTORY_QUERY_KEY, 'price', ALPH.symbol, { currency }],
```

Comment:
This key (and the next one) look like this:

<img width="618" alt="image" src="https://github.com/user-attachments/assets/fe27526e-42be-47c7-8ec9-2c8a6c8e78eb">

I am not sure yet what should go in an object  and what should go plain into the keys array. In the docs I see a lot `{ filter }` being used as an object. I guess it's not very important but I am curious to know pros/cons of the 2 approaches. Just food for thought.

Reply:
I think that the main point here is [this](https://tanstack.com/query/latest/docs/framework/react/guides/query-keys#query-keys-are-hashed-deterministically).